platform/plugins/pid/PidService.ts

```diff
@@ -6,35 +6,34 @@ import { PidFile } from './PidFile';
 import { LoggerFactory } from '../../logger';
 
 export class PidService {
-  private readonly pid$: Observable<PidFile | void>;
+  private readonly pid$: Observable<undefined>;
   private subscription?: Subscription;
 
-  constructor(pidConfig$: Observable<PidConfig | void>, logger: LoggerFactory) {
+  constructor(
+    pidConfig$: Observable<PidConfig | undefined>,
+    logger: LoggerFactory
+  ) {
     this.pid$ = pidConfig$
-      .map(
-        config =>
-          config !== undefined
-            ? new PidFile(process.pid, config, logger)
-            : undefined
-      )
       // Explanation of `switchMap`:
       // It's kinda like a normal `flatMap`, except it's producing observables
       // and you _only_ care about the latest observable it produced. It's
       // usually used if you need to control what happens both when you create
       // and when you're done with an observable, like here where we want to
       // write the pid file we receive a pid config, and delete it when we
       // receive new config values (or when we stop the pid service).
-      .switchMap(pid => {
-        if (pid === undefined) {
-          // If pid is not specified, we return an observable that does nothing
-          return new Observable<PidFile | void>(noop);
+      .switchMap(config => {
+        if (config === undefined) {
+          // If there is no pid config we return an observable that does nothing
+          return new Observable(noop);
         }
 
         // Otherwise we return an observable that writes the pid when
         // subscribed to and deletes it when unsubscribed (e.g. if new config
```

Comment:
Watching is "less safe" as you can be in intermediate states when it triggers. E.g. if you're moving stuff around in the config and hit save at some point (e.g. my IDE that saves whenever I cmd-tab away). It's better to let the user actively notify that the config is ready.